File: ideal-core/src/main/java/cn/idealframework2/idempotent/IdempotentHandlerFactory.java
package cn.idealframework2.idempotent;

import javax.annotation.Nonnull;
import java.time.Duration;

/**
 * @author 宋志宗 on 2022/10/13
 */
public interface IdempotentHandlerFactory {

  /**
   * 创建幂等处理器
   *
   * @param namespace 名称空间
   * @param expire    过期时间
   * @return 幂等处理器
   */
  @Nonnull
  IdempotentHandler create(@Nonnull String namespace, @Nonnull Duration expire);
}

File: ideal-core/src/main/kotlin/cn/idealframework2/idempotent/coroutine/IdempotentHandlerFactory.kt
package cn.idealframework2.idempotent.coroutine

import java.time.Duration

/**
 * @author 宋志宗 on 2022/10/13
 */
interface IdempotentHandlerFactory {

  /**
   * 创建幂等处理器
   *
   * @param namespace 名称空间
   * @param expire    过期时间
   * @return 幂等处理器
   */
  fun create(namespace: String, expire: Duration): IdempotentHandler
}
